mne/beamformer/_lcmv.py

```diff
@@ -211,44 +287,82 @@ def _apply_lcmv(data, info, tmin, forward, reg, noise_cov=None,
 
+    filters = dict(weights=W, data_cov=data_cov, noise_cov=noise_cov,
+                   whitener=whitener, weight_norm=weight_norm,
+                   pick_ori=pick_ori, ch_names=ch_names, proj=proj,
+                   is_ssp=is_ssp, vertices=vertno, is_free_ori=is_free_ori,
+                   nsource=forward['nsource'], src=deepcopy(forward['src']))
+
+    return filters
+
+
+def _subject_from_filter(filters):
+    """Get subject id from inverse operator."""
+    return filters['src'][0].get('subject_his_id', None)
+
+
+def _check_proj_match(info, filters):
+    """Check whether SSP projections in data and spatial filter match."""
+    proj_data, _, _ = make_projector(info['projs'],
+                                     filters['ch_names'])
+    if not np.array_equal(proj_data, filters['proj']):
+            raise ValueError('The SSP projections present in the data '
+                             'do not match the projections used when '
+                             'calculating the spatial filter.')
+
+
+def _apply_lcmv(data, filters, info, tmin, max_ori_out):
+    """Apply LCMV spatial filter to data for source reconstruction."""
+    if max_ori_out == 'abs':
+        warn('max_ori_out and the return of absolute values is deprecated and '
+             'will be removed in 0.16. Set it to "signed" to remove this '
+             'warning, this will return signed time series.',
+             DeprecationWarning)
+
     if isinstance(data, np.ndarray) and data.ndim == 2:
         data = [data]
         return_single = True
     else:
         return_single = False
 
-    subject = _subject_from_forward(forward)
+    W = filters['weights']
+
+    subject = _subject_from_forward(filters)
     for i, M in enumerate(data):
-        if len(M) != len(picks):
+        if len(M) != len(filters['ch_names']):
             raise ValueError('data and picks must have the same length')
 
         if not return_single:
             logger.info("Processing epoch : %d" % (i + 1))
 
-        # SSP and whitening
-        if info['projs']:
-            M = np.dot(proj, M)
-        if noise_cov is not None:
-            M = np.dot(whitener, M)
+        if filters['is_ssp']:
+            # check whether data and filter projs match
+            _check_proj_match(info, filters)
+            # apply projection
+            M = np.dot(filters['proj'], M)
+
+        if filters['whitener'] is not None:
+            M = np.dot(filters['whitener'], M)
 
         # project to source space using beamformer weights
-        if is_free_ori:
+        if filters['is_free_ori']:
             sol = np.dot(W, M)
             logger.info('combining the current components...')
             sol = combine_xyz(sol)
 
         else:
             # Linear inverse: do computation here or delayed
-            if M.shape[0] < W.shape[0] and pick_ori != 'max-power':
+            if (M.shape[0] < W.shape[0] and
+               filters['pick_ori'] != 'max-power'):
```

Comment:
this indentation looks wrong. It is one character more left than the line below.